### PR TITLE
Added menu view, summary and relationships validation

### PR DESCRIPTION
**Pull request by vprusa**
Also: https://polarion.engineering.redhat.com/polarion/#/project/JBossON4/workitem?id=JBossON4-9905
Sunil please verify that it does (intends to do) as it is described in the test case because im not sure if it does what it should at least for:
- Middleware Messagings->JMS Queues Summary page
- Middleware Messagings->JMS Topics Summary page 

About the TODOs and further refactoring:
At line 122 im not sure how would the page looked if something broke.. 
I think there is some red colored error message for broken links, but idk IDs/classes, please make a note if you know. Otherwise i will just add: 
- assert isTextOnPage("Error")
- assert isTextOnPage("error")
but that may not be enough...
There are some sleeps() that i did not know how to change:
- there is waiting for animation, the element is visible but not clickable, so either i would leave it as it is or do some try-click-until-successful method in utils that should be used just for animation

Thank you

### CODE REVIEW

**Comment by mattmahoneyrh** [comment]
Some items to address/fix:

-   delete_if_provider_present=False should be used w/ add_provider(), because there is no need to delete and then add the same provider prior to the test (delete/add provider is just a time waist).    

- Navigation should be using a "forced" type of navigate, and not hover navigation because of stability/dependability issues during both runs and across MIQ/CFME builds. Note: A hover type of navigation approach was first implemented, but was found be be quite unstable.
        Ex for navigating to Middleware Providers view:
        self.web_session.web_driver.get("{}//ems_middleware/show_list".format(self.web_session.MIQ_URL))
        assert self.ui_utils.waitForTextOnPage('Middleware Providers', 15)

-  Navigation to a Summary / Detailed view should be done similar to below (this approach has been very stable over many MIQ and CFME builds):
           Look at "validate_server_details() and in particular how the test cases gets the Servers list, and iterates through clicking on each row using ui_utils.click_on_row_containing_text()   

- Prior to merging PR, each of the following should really be in their respective test class:
  m.validate_middleware_providers()  test_providers.py
  m.validate_middleware_domains()    test_domains.py
  m.validate_middleware_servers()    test_servers.py
  m.validate_middleware_deployment() test_deployment.py
  m.validate_middleware_datasource() test_datasource.py
  m.validate_middleware_messaging()  test_messagings.py

File: views/menu.py [hunk 1]
```diff
@@ -20,26 +20,28 @@ def __init__(self, web_session):
...
+        pervious_url = self.web_driver.current_url
         self.web_session.web_driver.get(where)
-        counter = 15
-        while (current_url in self.web_driver.current_url and counter > 0):
+        counter = 5
+        while ((pervious_url is self.web_driver.current_url or not self.ui_utils.waitForTextOnPage(waitForText, 15)) and counter >= 0):
             self.ui_utils.sleep(2)
-            counter += 1
+            counter -= 1
+        assert counter > 0, "Waited long enough for redirect to {} and text '{}' to appear => failed ".format(where, waitForText)
```
Comment (mattmahoneyrh) [inline views/menu.py]: Consider using the timeout class,and take a look at existing examples.

**Comment by vprusa** [comment]
Now tested with Chrome, it was skipping validation because of using h1 for waitForText/Element. It needs to wait for text in table/th.

**Comment by mattmahoneyrh** [comment]
Test Assert failure: https://privatebin-it-iso.int.open.paas.redhat.com/?4d867a273539cb0a#1wHgITa756Avqwxe0tdzzkGCpTSl/6dyHIi4YLoAx8I=

**Comment by mattmahoneyrh** [comment]
Running the latest code, there continues to be a failure w/ Datasources: https://privatebin-it-iso.int.open.paas.redhat.com/?33e0da87cff5303e#CRQtKL4ZjwSviT8q2R2QuJr/logcdJf245l7ijH/M5U=

**Comment by mattmahoneyrh** [comment]
Here is the latest run with failure: https://privatebin-it-iso.int.open.paas.redhat.com/?e94a0474da64784c#qDUwOONPWw76t70o4CjFR+o0Vj0vIRXp6usWKUz+m8E=

**Comment by mattmahoneyrh** [comment]
One last minor request for enhancement, please consider using ui_utils.get_random_list() as a way to get a sample of maybe 2 or 3 tables items to validate. Otherwise, the test will validate all table entries, and for example Messagings and Datasources could results in too long of a test run when taking into consideration the number of tests that need to be run during an entire suite run.

**Comment by mattmahoneyrh** [comment]
what is the status of these tests? If ready, can you set up demo?

File: views/menu.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        return True
+
+    # validate each row in MW menu view, count=0 for all rows, otherwise 'count' used for random limit
+    def validate_rows(self, where, waitForText="", count=3):
+        self.ui_utils.waitForTextOnPage(waitForText, 15)
+        self.ui_utils.waitForElementOnPage(By.XPATH,"//th[contains(.,'"+waitForText+"')]/../../..//tbody/tr",15)
+        rows_count = len(self.ui_utils.get_list_table_as_elements()) if count == 0 else count
+        index = 0
+        while(index < rows_count):
+            table = self.ui_utils.get_list_table_as_elements() if count == 0 else self.ui_utils.get_random_list(self.ui_utils.get_list_table_as_elements(), count)
```
Comment (vprusa) [inline views/menu.py]: Randomized

**Pull request merged**